src/main/java/build/buildfarm/common/grpc/TracingMetadataUtils.java

```diff
@@ -35,6 +37,8 @@ private TracingMetadataUtils() {}
   private static final Context.Key<RequestMetadata> CONTEXT_KEY =
       Context.key("remote-grpc-metadata");
 
+  private static ArrayList<Property> capturedHeaders = new ArrayList<>();
```

Comment:
Specify variable type as bare semantics required - Iterable likely.

Also, what on earth? Why is this static? You will completely destroy any concurrent use, and never be guaranteed that 'headersFromCurrentContext' is valid with this workflow.